addons/mail/controllers/discuss/public_page.py

```diff
@@ -89,15 +94,16 @@ def _response_discuss_channel_invitation(self, store, channel):
         group_public_id = channel.group_public_id or channel.parent_channel_id.sudo().group_public_id
         if group_public_id and group_public_id not in request.env.user.groups_id:
             raise request.not_found()
-        guest_already_known = channel.env["mail.guest"]._get_guest_from_context()
         with replace_exceptions(UserError, by=NotFound()):
             # sudo: mail.guest - creating a guest and its member inside a channel of which they have the token
             __, guest = channel.sudo()._find_or_create_persona_for_channel(
                 guest_name=_("Guest"),
                 country_code=request.geoip.country_code,
                 timezone=request.env["mail.guest"]._get_timezone_from_request(request),
+                add_to_channel=False,
```

Comment:
Now that the user is not member, to_store conversion fail with access error. Steps to reproduce the issue:
- Install mail & im_livechat with demo data
- Go to the live chat channel and copy the invite link
- Open an incognito window, paste the link
- Forbidden

Reply:
Sorry for the delay.
I have updated the code to address the issue. Please have a look.
Thanks :slightly_smiling_face: